### PR TITLE
Fix SubPolygon and LinearRing for plotting

### PR DESCRIPTION
**Pull request by rafaqz**
This PR lets us use and plot the internal `SubPolygon` and `LinearRing` objects.

These are not part of any shapefile, but you can get these from `getgeom(multipolygon, 1)` or `getring(multipolygon, 1)` so we should be able to plot them, and all GeoInterface methods should be defined for them too.

### CODE REVIEW

File: src/polygons.jl [hunk 1]
```diff
@@ -15,8 +15,10 @@ LinearRing{P}(xy, z::Z, m::M) where {P,Z,M} = LinearRing{P,Z,M}(xy, z, m)
 LinearRing{P}(xy; z=nothing, m=nothing) where {P} = LinearRing{P}(xy, z, m)
 
 GI.isgeometry(::Type{<:LinearRing}) = true
+GI.is3d(::GI.LinearRingTrait, lr::LinearRing) = !isnothing(lr.z)
+GI.ismeasured(::GI.LinearRingTrait, lr::LinearRing) = isnothing(lr.m)
```
Comment (visr) [inline src/polygons.jl:19]:
```suggestion
GI.ismeasured(::GI.LinearRingTrait, lr::LinearRing) = !isnothing(lr.m)
```

**Review by visr — CHANGES_REQUESTED** [review]
Could use some tests.

**Comment by rafaqz** [comment]
"some tests" turned into quite a mission.

We were returning coordinates without `M` values in them so there were some inconsistencies with GeoInterface `ncoord`/`ismeasured`/`is3d`/`coordnames`/`coordinates` not lining up.

Adding some tests for that and for plotting. But now that M coordinates have M in them, they don't plot :frowning_face:

**Comment by rafaqz** [comment]
Also as noted in other threads, I cant reproduce these errors locally, so I cant finish this. If anyone can, please chime in with a fix.

**Comment by visr** [comment]
Just rebased this, and saw the same error as in CI.

The plot failed because there was a MethodError on GeoInterfaceRecipes._coordvecs. That method was added in https://github.com/JuliaGeo/GeoInterface.jl/commit/5520d68e3216847e07ee4b207571af0f455f1e84, but this commit has not yet been released. And you probably are running it on master :)

```julia
using GeoInterfaceRecipes
s = LinearRing{Point}(view(points, 1:4))
GeoInterfaceRecipes._coordvecs(GI.trait(s), s)
plot(s)
```

**Comment by rafaqz** [comment]
Aah ok. Guess we need to bump that then

**Pull request merged**